django_cassandra_engine/models.py

```diff
@@ -378,16 +381,35 @@ def order_by(self, *colnames):
                     *self._get_ordering_condition(colname))
                 )
             except query.QueryException as err:
+                if 'Can\'t order on' not in err.message:
+                    # if the exception isn't due to ordering, raise it up!
+                    raise err
+
                 msg = (
                     '.order_by() with column "{}" failed! '
-                    'falling back to unordered. '
+                    'falling back to ordering in python. '
                     'Exception was:\n{}'
                 ).format(colname, err.message)
                 warnings.warn(msg)
+                order_using_python = True
 
         clone = copy.deepcopy(self)
         clone._order.extend(conditions)
-        return clone
+
+        if order_using_python is True:
```

Comment:
TODO: refactor, test, add setting to explicitly trigger this behavior else raise.